[MTIA] Fix synchronize API

egienvalue opened the pull request:
Reviewed By: fenypatel99

Differential Revision: D58590313

aaronenyeshi reviewed (approved):
Please fix lintrunner issue, otherwise LGTM!

egienvalue committed:
[MTIA] Fix synchronize API (#128714)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/128714

Reviewed By: aaronenyeshi, fenypatel99

Differential Revision: D58590313
